utils.ps1

```diff
@@ -0,0 +1,298 @@
+# Need this variable as long as we support PS v2
+$ModuleBasePath = Split-Path $MyInvocation.MyCommand.Path -Parent
+
+# Store error records generated by stderr output when invoking an executable
+# This can be accessed from the user's session by executing:
+# PS> $m = Get-Module DbaTools
+# PS> & $m Get-Variable invokeErrors -ValueOnly
+$invokeErrors = New-Object System.Collections.ArrayList 256
+
+# General Utility Functions
+
+function Invoke-NullCoalescing {
+    $result = $null
+    foreach($arg in $args) {
+        if ($arg -is [ScriptBlock]) {
+            $result = & $arg
+        }
+        else {
+            $result = $arg
+        }
+        if ($result) { break }
+    }
+    $result
+}
+
+Set-Alias ?? Invoke-NullCoalescing -Force
+
+function Invoke-Utf8ConsoleCommand([ScriptBlock]$cmd) {
+    $currentEncoding = [Console]::OutputEncoding
+    $errorCount = $global:Error.Count
+    try {
+        # A native executable that writes to stderr AND has its stderr redirected will generate non-terminating
+        # error records if the user has set $ErrorActionPreference to Stop. Override that value in this scope.
+        $ErrorActionPreference = 'Continue'
+        if ($currentEncoding.IsSingleByte) {
+            [Console]::OutputEncoding = [Text.Encoding]::UTF8
+        }
+        & $cmd
+    }
+    finally {
+        if ($currentEncoding.IsSingleByte) {
+            [Console]::OutputEncoding = $currentEncoding
+        }
+
+        # Clear out stderr output that was added to the $Error collection, putting those errors in a module variable
+        if ($global:Error.Count -gt $errorCount) {
+            $numNewErrors = $global:Error.Count - $errorCount
+            $invokeErrors.InsertRange(0, $global:Error.GetRange(0, $numNewErrors))
+            if ($invokeErrors.Count -gt 256) {
+                $invokeErrors.RemoveRange(256, ($invokeErrors.Count - 256))
+            }
+            $global:Error.RemoveRange(0, $numNewErrors)
+        }
+    }
+}
+
+<#
+.SYNOPSIS
+    Configures your PowerShell profile (startup) script to import the DbaTools
+    module when PowerShell starts.
```

Comment:
Unfortunately, this deviates from standard PowerShell practice -- with PowerShell v3+ so long as dbatools is in the user's profile, it will dynamically load as needed. If anything, you could prompt the user if they'd like to modify their psmodulepath (HKCU, probably!) to include their git repo